vsphere-mixin/panels.libsonnet

```diff
@@ -0,0 +1,1472 @@
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('rotmhz')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      hostCPUUtilization:
+        commonlib.panels.memory.timeSeries.usagePercent.new(
+          'CPU utilization',
+          targets=[t.hostCPUUtilization],
+          description='The CPU utilization percentage of the ESXi host.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      hostMemoryUsage:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Memory usage',
+          targets=[t.hostMemoryUsage],
+          description='The amount of memory used by the ESXi host.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('mbytes')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      hostMemoryUtilization:
+        commonlib.panels.memory.timeSeries.usagePercent.new(
+          'Memory utilization',
+          targets=[t.hostMemoryUtilization],
+          description='The memory utilization percentage of the ESXi host.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table'),
+
+      hostModifiedMemory:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Modified memory',
+          targets=[t.hostModifiedMemoryBallooned, t.hostModifiedMemorySwapped],
+          description='The amount of memory that has been swapped or ballooned on the ESXi host.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('mbytes')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table')
+        + g.panel.timeSeries.options.legend.withPlacement('right'),
+
+      hostNetworkThroughputRate:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Avg network throughput rate',
+          targets=[t.hostNetworkTransmittedThroughputRate, t.hostNetworkReceivedThroughputRate],
+          description='The 20s average rate of data transmitted or received over the network of the ESXi host.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('KiBs')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table')
+        + g.panel.timeSeries.options.legend.withPlacement('bottom'),
+
+      hostPacketErrorRate:
+        commonlib.panels.generic.timeSeries.base.new(
+          'Avg packet errors',
+          targets=[t.hostPacketReceivedErrorRate, t.hostPacketTransmittedErrorRate],
+          description='The 20s average of received or transmitted packets dropped over the ESXi hosts network compared to the overall packets received or transmitted.'
+        )
+        + g.panel.timeSeries.standardOptions.withUnit('percent')
+        + g.panel.timeSeries.options.legend.withDisplayMode('table')
+        + g.panel.timeSeries.options.legend.withPlacement('bottom'),
```

Comment:
Similarly here, I'm not sure I'd be interested in anything but non-zero values showing up. @v-zhuravlev how have we addressed things like this in the past?

Reply:
@Dasomeone I've definitely had some panels where the queries had a != 0 in them. If an object is not consistently meeting criteria (in this case dropping packets) it can look spotty (you might just have a little positive blip on the graph) if we go this route though. Lines would only be consistent and smooth for hosts consistently dropping packets (maybe ok?)

Reply:
That's a good point for the error rate for sure. It probably wouldn't be quite so jarring if the panel type was swapped to a bar-chart, however?

Reply:
It's true...but it's also nice to see some history of errors 😄